tests/functional/adapter/upload_file_tests/test_upload_file.py

```diff
@@ -0,0 +1,109 @@
+import pytest
+from dbt.tests.util import (
+    run_dbt,
+    run_sql_with_adapter
+)
+import datetime
+import yaml
+
+_UPLOAD_FILE_SQL = """
+{% macro upload_file(local_file_path, database, table_schema, table_name) %}
+  {% do adapter.upload_file(local_file_path, database, table_schema, table_name, kwargs=kwargs) %}
+{% endmacro %}
+""".lstrip()
+
+
+class TestUploadFile:
+    @pytest.fixture(scope="class")
+    def schema(self):
+        return "upload_file"
+
+    @pytest.fixture(scope="class")
+    def macros(self):
+        return {
+            "upload_file.sql": _UPLOAD_FILE_SQL,
+        }
+
+    @staticmethod
+    def perform_uploaded_table_checks(table_schema, table_name, project):
```

Comment:
could we use dbt-core's util method `get_relation_columns` [here](https://github.com/dbt-labs/dbt-core/blob/065ab2ebc2104635770f49ca48bc434532d3ef41/core/dbt/tests/util.py#L471)?

Reply:
That simplifies it. I'll push a commit to add it.